src/lib/coda_intf/transition_frontier_components_intf.ml

```diff
@@ -233,15 +233,81 @@ module type Transition_handler_intf = sig
      and type transition_frontier_breadcrumb := transition_frontier_breadcrumb
 end
 
+module Best_tip_verification_result = struct
+  type 'external_transition t =
+    {root: 'external_transition; best_tip: 'external_transition}
```

Comment:
Can this type please be defined at a scope where `External_transition.t` is actually available to it (such as inside of the `Best_tip_prover_intf`)? This kind of "polymorphic record to remove a dependency cycle" has become a consistent issue in the code, and it is an anti-pattern I am attempting to avoid in it's entirety.